scripts/quests/bastok/Ayame_and_Kaede.lua

```diff
@@ -0,0 +1,197 @@
+                        quest:getVar(player, 'nmKilled') == 0 and
+                        not player:hasKeyItem(xi.ki.STRANGELY_SHAPED_CORAL) and
+                        npcUtil.popFromQM(player, npc, { korrolokaID.mob.KORROLOKA_LEECH_I, korrolokaID.mob.KORROLOKA_LEECH_II, korrolokaID.mob.KORROLOKA_LEECH_III }, { claim = true, hide = 0 })
+                    then
+                        return quest:messageSpecial(korrolokaID.text.SENSE_OF_FOREBODING)
+                    elseif quest:getVar(player, 'nmKilled') == 1 then
+                        npcUtil.giveKeyItem(player, xi.ki.STRANGELY_SHAPED_CORAL)
+                        quest:setVar(player, 'nmKilled', 0)
+                        return quest:noAction()
+                    end
+                end,
+            },
+
+            ['Korroloka_Leech'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        GetMobByID(korrolokaID.mob.KORROLOKA_LEECH_I):isDead() and
+                        GetMobByID(korrolokaID.mob.KORROLOKA_LEECH_II):isDead() and
+                        GetMobByID(korrolokaID.mob.KORROLOKA_LEECH_III):isDead() and
+                        quest:getVar(player, 'Prog') == 2
+                    then
+                        quest:setVar(player, 'nmKilled', 1)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Ensetsu'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        not player:hasKeyItem(xi.ki.STRANGELY_SHAPED_CORAL)
+                    then
+                        return quest:event(242)
+                    elseif player:hasKeyItem(xi.ki.STRANGELY_SHAPED_CORAL) then
+                        return quest:progressEvent(245)
+                    elseif player:hasKeyItem(xi.ki.SEALED_DAGGER) then
+                        return quest:progressEvent(246, xi.ki.SEALED_DAGGER)
+                    end
+                end,
+            },
+
+            ['Kagetora'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') > 2 then
+                        return quest:event(244)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [245] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [246] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.SEALED_DAGGER)
+                        player:unlockJob(xi.job.NIN)
+                        player:messageSpecial(portBastokID.text.KEYITEM_LOST, xi.ki.SEALED_DAGGER) -- not able to do 2 return message specials?
```

Comment:
nah as soon as you use a return, it'll exit the quest framework and quit reading lines. Just any other regular functions:

bool flag = true

if flag == true then
    return flag
end

return false

As you see the false will never be returned or even reached